index.js

```diff
@@ -373,6 +373,16 @@ function doTransform(checker, targetVersion, k) {
         ts.unescapeLeadingUnderscores(member.name.escapedText),
         /*hasTrailingNewline*/ false
       );
+    } else if (semver.lt(targetVersion, "4.7.0") && ts.isTypeParameterDeclaration(n)) {
+      return ts.factory.createTypeParameterDeclaration(
+        n.modifiers &&
+          n.modifiers.filter(
+            modifier => modifier.kind !== ts.SyntaxKind.InKeyword && modifier.kind !== ts.SyntaxKind.OutKeyword
+          ),
```

Comment:
```suggestion
        n.modifiers?.filter(
          modifier => modifier.kind !== ts.SyntaxKind.InKeyword && modifier.kind !== ts.SyntaxKind.OutKeyword
        ),
```

Reply:
That's what I've done at the beginning, but it turns out that Babel (that is used by Jest) is not configured to handle this:
<details>
<summary>Error output</summary>

```
> downlevel-dts@0.9.0 test
> jest

 FAIL  ./index.test.js
  ● Test suite failed to run

    SyntaxError: /downlevel-dts/index.js: Support for the experimental syntax 'optionalChaining' isn't currently enabled (378:20):

      376 |     } else if (semver.lt(targetVersion, "4.7.0") && ts.isTypeParameterDeclaration(n)) {
      377 |       return ts.factory.createTypeParameterDeclaration(
    > 378 |         n.modifiers?.filter(
          |                    ^
      379 |           modifier => modifier.kind !== ts.SyntaxKind.InKeyword && modifier.kind !== ts.SyntaxKind.OutKeyword
      380 |         ),
      381 |         n.name,

    Add @babel/plugin-proposal-optional-chaining (https://git.io/vb4Sk) to the 'plugins' section of your Babel config to enable transformation.

      at Parser.raise (node_modules/@babel/parser/src/parser/location.js:41:63)
      at Parser.expectPlugin (node_modules/@babel/parser/src/parser/util.js:165:18)
      at Parser.parseSubscript (node_modules/@babel/parser/src/parser/expression.js:586:12)
      at Parser.parseSubscripts (node_modules/@babel/parser/src/parser/expression.js:555:19)
      at Parser.parseExprSubscripts (node_modules/@babel/parser/src/parser/expression.js:540:17)
      at Parser.parseMaybeUnary (node_modules/@babel/parser/src/parser/expression.js:507:21)
      at Parser.parseExprOps (node_modules/@babel/parser/src/parser/expression.js:275:23)
      at Parser.parseMaybeConditional (node_modules/@babel/parser/src/parser/expression.js:230:23)
      at Parser.parseMaybeAssign (node_modules/@babel/parser/src/parser/expression.js:180:21)
      at Parser.parseExprListItem (node_modules/@babel/parser/src/parser/expression.js:2083:18)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.607s, estimated 9s
Ran all test suites.
```
</details>

I could fix this as part of this PR but I've wanted to keep it focused. If you prefer this to be fixed here - or in a separate PR so we could later rebase this one here, let me know.

Reply:
I agree with keeping this PR focussed. I'll update myself after upgrading Jest.